src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1898,87 +1898,126 @@ impl Cage {
     }
 
     //------------------------------------FCNTL SYSCALL------------------------------------
+    
+    //fcntl performs operations, like returning or setting file status flags,
```

Comment:
Looks great!

@yashaswi2000 , do you want to have the rustdoc changes be done in these edits or a different set of PRs?  I do think that for new PRs, we likely should include them...

Reply:
I will refactor these first 3 PRs. and any new ones coming in will follow the template I will share with them.